### ISSUE TITLE
KeyProvider23.java line 80

### ISSUE THREAD

**Issue by owdeveloper** [aws-amplify/aws-sdk-android#1109]
**Describe the bug**
We're getting a very high volume of AWS library crashes in Firebase Crashlytics, could you check?

**Which AWS service(s) are affected?**
Android SDK 2.13.5
`implementation "com.amazonaws:aws-android-sdk-core:2.13.5"`
`implementation "com.amazonaws:aws-android-sdk-cognitoidentityprovider:2.13.5"`
`implementation "com.amazonaws:aws-android-sdk-appsync:2.10.0"`

**Screenshots**
![image](https://user-images.githubusercontent.com/6125289/62375942-ddb3ce80-b515-11e9-94bd-039e4c625774.png)
![image](https://user-images.githubusercontent.com/6125289/62375950-ea382700-b515-11e9-8d8b-45a5fd2314f7.png)

**Additional context**
Firebase Crashlytics.
```
Fatal Exception: java.lang.RuntimeException: Unable to create application com.pontomobi.smiles.SmilesApplication: java.lang.IllegalStateException: Error in initializing the CognitoCachingCredentialsProvider. 
       at android.app.ActivityThread.handleBindApplication + 6499(ActivityThread.java:6499)
       at android.app.ActivityThread.access$1800 + 229(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage + 1887(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 7406(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1230(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main + 1120(ZygoteInit.java:1120)
```
```
Caused by java.lang.IllegalStateException: Error in initializing the CognitoCachingCredentialsProvider. 
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 434(CognitoCachingCredentialsProvider.java:434)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1037(Instrumentation.java:1037)
       at android.app.ActivityThread.handleBindApplication + 6496(ActivityThread.java:6496)
       at android.app.ActivityThread.access$1800 + 229(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage + 1887(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 7406(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1230(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main + 1120(ZygoteInit.java:1120)
```
```
Caused by java.lang.IllegalStateException: java.security.UnrecoverableKeyException: Failed to obtain information about key
       at com.amazonaws.internal.keyvaluestore.KeyProvider23.getKey + 84(KeyProvider23.java:84)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.setPersistenceEnabled + 131(AWSKeyValueStore.java:131)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.<init> + 108(AWSKeyValueStore.java:108)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 425(CognitoCachingCredentialsProvider.java:425)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1037(Instrumentation.java:1037)
       at android.app.ActivityThread.handleBindApplication + 6496(ActivityThread.java:6496)
       at android.app.ActivityThread.access$1800 + 229(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage + 1887(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 7406(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1230(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main + 1120(ZygoteInit.java:1120)
```
```
Caused by java.security.UnrecoverableKeyException: Failed to obtain information about key
       at android.security.keystore.AndroidKeyStoreProvider.loadAndroidKeyStoreSecretKeyFromKeystore + 275(AndroidKeyStoreProvider.java:275)
       at android.security.keystore.AndroidKeyStoreSpi.engineGetKey + 97(AndroidKeyStoreSpi.java:97)
       at java.security.KeyStore.getKey + 253(KeyStore.java:253)
       at com.amazonaws.internal.keyvaluestore.KeyProvider23.getKey + 80(KeyProvider23.java:80)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.setPersistenceEnabled + 131(AWSKeyValueStore.java:131)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.<init> + 108(AWSKeyValueStore.java:108)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 425(CognitoCachingCredentialsProvider.java:425)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1037(Instrumentation.java:1037)
       at android.app.ActivityThread.handleBindApplication + 6496(ActivityThread.java:6496)
       at android.app.ActivityThread.access$1800 + 229(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage + 1887(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 7406(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1230(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main + 1120(ZygoteInit.java:1120)
```
```
Caused by android.security.KeyStoreException: -49
       at android.security.KeyStore.getKeyStoreException + 945(KeyStore.java:945)
       at android.security.keystore.AndroidKeyStoreProvider.loadAndroidKeyStoreSecretKeyFromKeystore + 275(AndroidKeyStoreProvider.java:275)
       at android.security.keystore.AndroidKeyStoreSpi.engineGetKey + 97(AndroidKeyStoreSpi.java:97)
       at java.security.KeyStore.getKey + 253(KeyStore.java:253)
       at com.amazonaws.internal.keyvaluestore.KeyProvider23.getKey + 80(KeyProvider23.java:80)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.setPersistenceEnabled + 131(AWSKeyValueStore.java:131)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.<init> + 108(AWSKeyValueStore.java:108)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 425(CognitoCachingCredentialsProvider.java:425)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1037(Instrumentation.java:1037)
       at android.app.ActivityThread.handleBindApplication + 6496(ActivityThread.java:6496)
       at android.app.ActivityThread.access$1800 + 229(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage + 1887(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 7406(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1230(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main + 1120(ZygoteInit.java:1120)
```

**Comment by palpatim** [comment]
Working on a fix for this now as part of #937

**Comment by palpatim** [comment]
We released a fix for this on [2.15.1](https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.15.1). Please let us know if you have any questions or continue to see crashes related to retrieving encrypted data.
